internal/captain/command.go

```diff
@@ -260,14 +267,30 @@ func (c *Command) Flags() []*Flag {
 	return c.flags
 }
 
-func (c *Command) Executor() Executor {
+func (c *Command) ExecuteFunc() ExecuteFunc {
 	return c.execute
 }
 
 func (c *Command) Arguments() []*Argument {
 	return c.arguments
 }
 
+func (c *Command) SetInterceptChain(fns ...InterceptFunc) {
+	c.interceptChain = fns
+}
+
+func (c *Command) interceptFunc() InterceptFunc {
+	return func(fn ExecuteFunc) ExecuteFunc {
+		for i := len(c.interceptChain) - 1; i >= 0; i-- {
```

Comment:
What's wrong with `for _, intercept := range c.interceptChain` ?

This is a nit really. Just seems needlessly verbose.

Reply:
The loop must iterate in reverse or the nesting order will be backward.